source.go

```diff
@@ -334,12 +337,12 @@ func (s *Source) pageToRecord(ctx context.Context, page *notion.Page, children n
 		return sdk.Record{}, fmt.Errorf("failed getting payload: %w", err)
 	}
 
-	return sdk.Record{
-		Metadata:  nil,
-		CreatedAt: time.Now(),
-		Key:       sdk.RawData(page.ID),
-		Payload:   payload,
-	}, nil
+	return sdk.Util.Source.NewRecordCreate(
+		nil,
```

Comment:
The position will be set in `Source.nextPage` once we're sure it's safe to be saved.